### PR TITLE
explicitly specify url parameter in post

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

File: rubin_jupyter_utils/helpers/functions.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def parse_access_token(endpoint=None, tokenfile=None, token=None, timeout=15):
     # Endpoint is constant in an ArgoCD-deployed cluster
     if not endpoint:
         endpoint = "http://gafaelfawr-service.gafaelfawr:8080/auth/analyze"
-    resp = requests.post(endpoint, data={"token": token}, timeout=timeout)
+    resp = requests.post(url=endpoint, data={"token": token}, timeout=timeout)
```
Comment (SimonKrughoff) [inline rubin_jupyter_utils/helpers/functions.py:204]: I'm just curious why this is necessary, it's a positional argument according to the API docs

**Review by SimonKrughoff — APPROVED** [review]

**Comment by rra** [comment]
As with Simon, I'm not sure what problem this is solving; I don't think it should matter?

Unrelated to this change, but since I'm looking at this code, it's unfortunate that it's talking directly to the underlying Gafaelfawr service, since ideally I'd like to use a `NetworkPolicy` to restrict access to Gafaelfawr except through the ingress. Otherwise, people in RSP pods can mint arbitrary internal and notebook tokens for themselves, which while not a huge deal is not the intended behavior. Does this code have access to the real external hostname for the RSP deployment anywhere so that it can go through the ingress?

File: rubin_jupyter_utils/helpers/functions.py (hunk 1 above)
Comment (athornton) [reply rubin_jupyter_utils/helpers/functions.py:204]: Yeah, I dunno either, but give it a try as-is on data-dev!

**Comment by athornton** [comment]
> As with Simon, I'm not sure what problem this is solving; I don't think it should matter?
> 

Yeah, neither did I, but try using parse_user_token on data-dev, and you'll see that it's deciding the token is the endpoint.  Adding the url= parm makes it work again.

> Unrelated to this change, but since I'm looking at this code, it's unfortunate that it's talking directly to the underlying Gafaelfawr service, since ideally I'd like to use a `NetworkPolicy` to restrict access to Gafaelfawr except through the ingress. Otherwise, people in RSP pods can mint arbitrary internal and notebook tokens for themselves, which while not a huge deal is not the intended behavior. Does this code have access to the real external hostname for the RSP deployment anywhere so that it can go through the ingress?

I'll take a look.  I think that is exposed in the RSP environment.

**Comment by athornton** [comment]
Yeah I think this was PEBCAK.  I'll take up the FQDN thing separately.

**Pull request closed without merging**